### PR TITLE
fix: Enhancement in color page token

### LINKED ISSUE #4110: Build color page enhancements
Based on several conversations over the last few months, we have concluded the color token page has become too large and hard to navigate. We had an [original color page redesign](https://github.com/carbon-design-system/carbon-website/issues/2476) that was put on ice in favor of a platform that was never fully built, and the time to AEM is too far out to not make this page easier for users.  

We've put together a new design concept using only the components that already exist in Gatsby to make a better token experience. The enhancements include: 
- Multiple H2 sections for Core, Component, and AI tokens 
- Adding H3 anchor links inline with the H3 token headers in tables. 
- Adding H3 mini-anchor link sections for easier navigation. 

[Figma file](https://www.figma.com/design/7ubdj4yYwoI6IAuEmp4vB8/Color-page-updates?node-id=1-8394&t=fl9f2AETbZbPyJo3-1)

color page update

### PR DESCRIPTION
**Pull request by riddhybansal**
Closes #4110 

Color Page Enhancements

#### Changelog

**New**

-Multiple H2 sections for Core, Component, and AI tokens
-Adding H3 anchor links inline with the H3 token headers in tables.
-Adding H3 mini-anchor link sections for easier navigation.
-Addition of a table under component tokens called "notification"

### CODE REVIEW

**Comment by riddhybansal** [comment]
Hey reviewer , Please review the notification table and there hex code as this is newly added. Please verify 2 links added in the the description of component tokens and ai tokens are correct or not. Please check if the tokens are correctly categorised and navigating to the correct table.

File: src/pages/elements/color/tokens.mdx [hunk 1]
```diff
@@ -8,6 +8,83 @@ tabs: ['Overview', 'Usage', 'Tokens', 'Code']
...
+</AnchorLinks>
+
+## Core Tokens
+
+Core tokens are global colors that are used across components. They are named
+and grouped by the common UI element that they are applied to.
+
+<AnchorLinks>
+
+<AnchorLink>Background</AnchorLink>
```
Comment (aagonzales) [inline src/pages/elements/color/tokens.mdx:33]: These anchor links under the H2 sections are coming in at the larger size. Can we make them smaller like the anchor link on the [2x Grid](https://carbondesignsystem.com/elements/2x-grid/overview/#2x-grid-fundamentals) page?

File: src/pages/elements/color/tokens.mdx [hunk 2]
```diff
@@ -8,6 +8,83 @@ tabs: ['Overview', 'Usage', 'Tokens', 'Code']
...
+
+<ColorTokenTable type="component" />
+
+## AI Tokens
+
+To accommodate for the new AI styles, we have introduced a new suite of color
+tokens that can be found within the main Carbon themes. These tokens should only
+be used when building custom AI components, variants, or instances in your UI.
+For more information on how to use the AI token, see the
+
```
Comment (aagonzales) [inline src/pages/elements/color/tokens.mdx:76]: There is a weird line break here: 
image

File: src/pages/elements/color/tokens.mdx [hunk 3]
```diff
@@ -8,6 +8,83 @@ tabs: ['Overview', 'Usage', 'Tokens', 'Code']
...
+<ColorTokenTable type="core" />
+
+## Component Tokens
+
+Some components have their own specific color tokens, known as component tokens.
+They represent the properties associated with a particular component. They are
+not global tokens like the core tokens and should never be used for anything
+other than their own component. See the
+
```
Comment (aagonzales) [inline src/pages/elements/color/tokens.mdx:55]: There is a weird gap here 
image

File: src/pages/elements/color/tokens.mdx [hunk 4]
```diff
@@ -8,6 +8,83 @@ tabs: ['Overview', 'Usage', 'Tokens', 'Code']
...
+<a href="https://github.com/carbon-design-system/carbon/blob/main/packages/styles/docs/sass.md#component-tokens">
+  developer documentation{' '}
+</a> for more information.
```
Comment (alisonjoseph) [inline src/pages/elements/color/tokens.mdx:58]: Should be able to switch this to markdown syntax to fix the line break vs html. 
`[developer documentation](https://github.com/carbon-design-system/carbon/blob/main/packages/styles/docs/sass.md#component-tokens)`

File: src/pages/elements/color/tokens.mdx [hunk 5]
```diff
@@ -8,6 +8,83 @@ tabs: ['Overview', 'Usage', 'Tokens', 'Code']
...
+<AnchorLink>AI Tokens</AnchorLink>
+
+</AnchorLinks>
+
+## Core Tokens
+
+Core tokens are global colors that are used across components. They are named
+and grouped by the common UI element that they are applied to.
+
+<AnchorLinks>
```
Comment (alisonjoseph) [inline src/pages/elements/color/tokens.mdx:31]:
```suggestion
<AnchorLinks small>
```

**Review by aagonzales — APPROVED** [review]
Looks good to me!

File: src/pages/elements/color/tokens.mdx [hunk 6]
```diff
@@ -8,6 +8,77 @@ tabs: ['Overview', 'Usage', 'Tokens', 'Code']
...
+</AnchorLinks>
+
+<ColorTokenTable type="component" />
+
+## AI Tokens
+
+To accommodate for the new AI styles, we have introduced a new suite of color
+tokens that can be found within the main Carbon themes. These tokens should only
+be used when building custom AI components, variants, or instances in your UI.
+For more information on how to use the AI token, see the
```
Comment (alisonjoseph) [inline src/pages/elements/color/tokens.mdx:72]:
```suggestion
For more information on how to use the AI tokens, see the
```

File: src/pages/elements/color/tokens.mdx [hunk 7]
```diff
@@ -8,6 +8,77 @@ tabs: ['Overview', 'Usage', 'Tokens', 'Code']
...
+
+<ColorTokenTable type="core" />
+
+## Component Tokens
+
+Some components have their own specific color tokens, known as component tokens.
+They represent the properties associated with a particular component. They are
+not global tokens like the core tokens and should never be used for anything
+other than their own component. See the
+[developer documentation](https://github.com/carbon-design-system/carbon/blob/main/packages/styles/docs/sass.md#component-tokens)
```
Comment (alisonjoseph) [inline src/pages/elements/color/tokens.mdx:55]:
```suggestion
other than their own component. For more information on how to use component tokens, see the
[developer documentation](https://github.com/carbon-design-system/carbon/blob/main/packages/styles/docs/sass.md#component-tokens).
```

**Review by alisonjoseph — CHANGES_REQUESTED** [review]
Couple super minor copy edits, but this looks great! 🔥

**Comment by riddhybansal** [comment]
> Couple super minor copy edits, but this looks great! 🔥

OOPs small ones :(

**Review by alisonjoseph — APPROVED** [review]
LGTM! I think you just need to run `yarn format` to get CI to pass

**Pull request merged**